packages/headless/src/controllers/search-box/headless-search-box.ts

```diff
@@ -205,11 +205,16 @@ export function buildSearchBox(
 
     selectSuggestion(value: string) {
       dispatch(selectQuerySuggestion({id, expression: value}));
-      performSearch(logQuerySuggestionClick({id, suggestion: value}));
+      performSearch(logQuerySuggestionClick({id, suggestion: value})).then(
+        () => {
+          dispatch(clearQuerySuggest({id}));
+        }
+      );
     },
 
     submit() {
       performSearch(logSearchboxSubmit());
+      dispatch(clearQuerySuggest({id}));
```

Comment:
Why is there no promise chain for this clear compared to line 208?

Reply:
Not needed here. `logSearchBoxSubmit()` does not need to read partial queries/completion arrays to do it's job properly.

We can clear "right away" without waiting.